src/freenas/usr/local/lib/middlewared_truenas/plugins/truenas.py

```diff
@@ -90,6 +90,8 @@ async def get_chassis_hardware(self):
 
         chassis = await run('dmidecode', '-s', 'system-product-name', check=False)
         chassis = chassis.stdout.decode(errors='ignore').split('\n', 1)[0].strip()
+        if chassis == 'TRUENAS-MINI':
+            return 'TRUENAS-UNKNOWN'
         if chassis.startswith(('TRUENAS-M', 'TRUENAS-X', 'TRUENAS-Z')):
```

Comment:
Do we use the exact version M40 X10, X20 anywhere?

Reply:
I still don't know why we wouldn't just use `failover.hardware == 'ECHOWARP'`

Reply:
I can understand that ECHOWARP may make it more difficult for people reading the code, its not directly clear it references M-series, its "obscure"

Reply:
But it's the proper check. Write another method in `failover` plugin that returns True/False.

def is_mseries(self):
    if failover.hardware == 'ECHOWARP':
        return True
    chassis = truenas.get_chassis_hardware
    if chassis.startswith('TRUENAS-M') and 'MINI' not in chassis:
        return True
    return False

Reply:
Its redundant, we already have the chassis hardware for this. We dont need is_xseries, is_zseries and is_mseries methods. One is enough

Reply:
@william-gr what have we decided then? `==` or `startswith`?

Reply:
I am not a fan of `failover.hardware == 'ECHOWARP'` either

Reply:
@themylogin do we check for specific hardware anything using that method? If we do then we keep with startswith

Reply:
@william-gr yes, we do